### PR TITLE
Switched land values to nan for plotting and metric calculation

### PR DESCRIPTION
**Pull request by suryadheeshjith**
- RMSE during train/eval will increase because of this change as we are considering land as nans and not 0s anymore
- Removed some redundant normalization functions
- Normalization functions weirdly changed the data type from float32 to float64 so I converted the final result back to float32
- Fixed plotting to handle Nan values, now land looks a lot better in the maps
- Added tests for aggregator metric weighted_mean and normalization functions

### CODE REVIEW

File: src/ocean_emulators/aggregator/metrics.py [hunk 1]
```diff
@@ -11,6 +11,35 @@ def weighted_mean(
 ) -> torch.Tensor:
     """Computes the weighted mean across the specified list of dimensions.
 
+    Args:
+        tensor: torch.Tensor
```
Comment (jder) [inline src/ocean_emulators/aggregator/metrics.py:18]: How about documenting here how we treat NaNs in the tensor?

File: src/ocean_emulators/aggregator/metrics.py [hunk 2]
```diff
@@ -11,6 +11,35 @@ def weighted_mean(
...
+
+def weighted_mean_with_nan_as_zero(
+    tensor: torch.Tensor,
+    weights: Optional[torch.Tensor] = None,
+    dim: tuple[int, ...] = (-2, -1),
+    keepdim: bool = False,
+) -> torch.Tensor:
+    """This function used purely for testing.
+
+    Similar to weighted_mean, but treats assumes input with Nans as zeros.
```
Comment (jder) [inline src/ocean_emulators/aggregator/metrics.py]: Any reason to keep this around? Seems like we don't use it. If we do want to keep it around, I'm not sure what it means to "treats assumes input with Nans as zeros". How about "assumes there are no NaNs in tensor" or just "returns NaN if there are any NaNs in tensor"? (And maybe change the function name to match eg `weighted_mean_no_nans`?)

File: src/ocean_emulators/utils/data.py [hunk 3]
```diff
@@ -370,9 +364,12 @@ def normalize_tensor_prognostic(
         norm = (data - tensor_mean) / tensor_std
         if fill_nan:
             norm = norm.nan_to_num(nan=fill_value)
+        norm = norm.to(data.dtype)
         return norm
 
-    def unnormalize_tensor_prognostic(self, data: torch.Tensor) -> torch.Tensor:
+    def unnormalize_tensor_prognostic(
+        self, data: torch.Tensor, apply_nan=False
```
Comment (jder) [inline src/ocean_emulators/utils/data.py]: How about `mask_value: float = 0` and then people can pass `mask_value = float("nan")` if they want? Otherwise it's a little unclear to me from the name what "apply"ing a NaN is. Or could add docs clarifying?

File: tests/test_aggregator_metrics.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    assert torch.allclose(result_with_nan, expected_with_nan)
+
+    # Test case 3: Without weights
+    result_no_weights = weighted_mean(tensor)
+    expected_no_weights = tensor.nanmean(dim=(-2, -1))
+    assert torch.allclose(result_no_weights, expected_no_weights)
+
+    # Test case 4: With keepdim=True
+    result_keepdim = weighted_mean(tensor, weights, keepdim=True)
+    assert result_keepdim.shape == (1, 1)
```
Comment (jder) [inline tests/test_aggregator_metrics.py:24]: Thanks for the tests! It's a little funny to me when the test code and the "expected" code are basically the same implementation. What about doing something like this where we know the answer for test & the below tests?

```
    # each value * weight should == 1
    tensor = torch.tensor([[1.0, 2.0], [4.0, 8.0]])
    weights = torch.tensor([[1.0, 0.5], [0.25, 0.125]])
    result = weighted_mean(tensor, weights)
    assert torch.allclose(result, 4.0)
```

File: tests/test_aggregator_metrics.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+
+def test_area_weighted_mean():
+    data = torch.tensor([[1.0, 2.0], [3.0, 4.0]])
+    area_weights = torch.tensor([[0.1, 0.2], [0.3, 0.4]])
+    result = area_weighted_mean(data, area_weights)
+    expected = weighted_mean(data, area_weights)
+    assert torch.allclose(result, expected)
```
Comment (jder) [inline tests/test_aggregator_metrics.py]: I'm not sure what this is testing. `area_weighted_mean` just calls `weighted_mean` internally.

File: tests/test_utils_data.py [hunk 6]
```diff
@@ -123,11 +126,61 @@ def test_compute_anomalies():
...
+            "var_1": (["lat", "lon"], [[1.0]]),
+            "var_2": (["lat", "lon"], [[2.0]]),
+        },
+        coords={"lat": [0], "lon": [0]},
+    )
+
+    # Create test wet mask
+    wet_mask = torch.tensor([[1.0, 0.0], [0.0, 1.0]])
+    # Initialize Normalize instance
+    with MultitonScope():
```
Comment (jder) [inline tests/test_utils_data.py:161]: This makes sense to me! Might have a comment saying this is creating a non-global Normalize instance so it doesn't affect other tests.

**Review by jder — APPROVED** [review]
Just some suggestions for minor cleanups from me, generally looks good!

File: tests/test_utils_data.py (hunk 6 above)
Comment (alxmrs) [reply tests/test_utils_data.py:161]: I'm going to borrow this pattern :)

File: src/ocean_emulators/aggregator/metrics.py (hunk 2 above)
Comment (alxmrs) [reply src/ocean_emulators/aggregator/metrics.py]: 🐑  Further on this: instead of a new method, could the Nan support be handled as a flag? The two routines seems really similar.

**Review by alxmrs — APPROVED** [review]
LGTM too :)

File: src/ocean_emulators/aggregator/metrics.py (hunk 2 above)
Comment (suryadheeshjith) [reply src/ocean_emulators/aggregator/metrics.py]: So this function is what weighted_mean previously was and is actually an incorrect function when we pass in Nans to the data (which we didn't before since the land cells were 0s and not Nan). Maybe we could move this into the test file instead?

File: tests/test_aggregator_metrics.py (hunk 5 above)
Comment (suryadheeshjith) [reply tests/test_aggregator_metrics.py]: I can remove this

File: tests/test_utils_data.py (hunk 6 above)
Comment (suryadheeshjith) [reply tests/test_utils_data.py:161]: Okay, I was hoping this was how we wanted to test singletons!

File: src/ocean_emulators/aggregator/metrics.py (hunk 2 above)
Comment (suryadheeshjith) [reply src/ocean_emulators/aggregator/metrics.py]: Actually, I will just remove this function!

File: src/ocean_emulators/utils/data.py (hunk 3 above)
Comment (suryadheeshjith) [reply src/ocean_emulators/utils/data.py]: Renamed to fill_value and added some documentation
